modules/quant_purger/src/Plugin/Purge/Purger/QuantPurgeBase.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace Drupal\quant_purger\Plugin\Purge\Purger;
+
+use Drupal\Core\Utility\Token;
+use GuzzleHttp\ClientInterface;
+use Drupal\purge\Plugin\Purge\Purger\PurgerBase;
+use Drupal\purge\Plugin\Purge\Purger\PurgerInterface;
+use Drupal\quant_purger\Entity\QuantPurgeSettings;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\purge\Plugin\Purge\Invalidation\InvalidationInterface;
+
+/**
+ * Abstract base class for HTTP based configurable purgers.
+ */
+abstract class QuantPurgeBase extends PurgerBase implements PurgerInterface {
+
+  /**
+   * The Guzzle HTTP client.
+   *
+   * @var \GuzzleHttp\Client
+   */
+  protected $client;
+
+  /**
+   * The settings entity holding all configuration.
+   *
+   * @var \Drupal\quant_purger\Entity\QuantPurgeSettings
+   */
+  protected $settings;
+
+  /**
+   * The settings entity holding the Quant API configuration.
+   */
+  protected $api_settings;
+
+  /**
+   * The token service.
+   *
+   * @var \Drupal\Core\Utility\Token
+   */
+  protected $token;
+
+  /**
+   * Constructs the HTTP purger.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   An HTTP client that can perform remote requests.
+   * @param \Drupal\Core\Utility\Token $token
+   *   The token service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ClientInterface $http_client, Token $token) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->settings = QuantPurgeSettings::load($this->getId());
+    // Note: We use the Quant HTTP client rather than the generic Guzzle client.
+    $this->client = \Drupal::service('quant_api.client');
```

Comment:
I think this can be injected with the `create` method rather than accessing `\Drupal::service` here.